src/intelligence_layer/core/model.py

```diff
@@ -406,3 +408,135 @@ def to_instruct_prompt(
         return self.INSTRUCTION_PROMPT_TEMPLATE.to_rich_prompt(
             instruction=instruction, input=input, response_prefix=response_prefix
         )
+
+
+class Message(BaseModel):
+    role: Literal["system", "user", "assistant"]
+    content: str
+
+
+class ChatModel(ABC, AlephAlphaModel):
+    RECOMMENDED_MODELS: ClassVar[list[str]] = []
```

Comment:
we should probably refactor this up to the AA-Model

Reply:
or to some other abstractions, as these things are quite duplicated with the ControlModel

Reply:
What exactly is duplicated?
They serve two somewhat disctinct purposes. The only duplicate thing I see is the abstract eot_token function, but I also wouldn't move this up, because that would make AlephAlphaModel abstract